src/refract/element.es6

```diff
@@ -0,0 +1,92 @@
+/*
+ * Base elements for working with refract shorthand, which is composed of
+ * tuples with the following values:
+ *
+ * [name, metadata, attributes, contents]
+ *
+ * For example:
+ *
+ * ['resource', {'title': 'Question'}, {'href': '/questions'},
+ *   ['transition', {}, {}, ...],
+ *   ...
+ * ]
+ *
+ * https://github.com/refractproject/refract-spec
+ */
+const rawElement = Symbol('element');
+
+export default class RefractElement {
+  constructor(element=['', {}, {}, null]) {
+    this[rawElement] = element;
+    this.name = element[0];
+
+    if (element[1]) {
+      this.title = element[1].title;
+      // TODO: Render the markdown?
+      this.description = element[1].description;
+    }
+  }
+
+  /*
+   * Returns `true` if there is a `class` attribute list and `name` is present
+   * in the list.
+   */
+  hasClass(name) {
+    return this[rawElement][1] && this[rawElement][1].class &&
+           this[rawElement][1].class.indexOf &&
+           this[rawElement][1].class.indexOf(name) !== -1;
+  }
+
+  /*
+   * Gather the contents of an element, optionally selecting a specific
+   * key (if it is a hash) and filtering by element name or class name.
+   * Multiple filters are treated as an AND operation, e.g. return all
+   * 'category' elements with a class of 'resourceGroup'.
+   *
+   * options = {
+   *   className,   // Filter by element[1].class.
+   *   key,         // Select this key (useful if the element is a hash).
+   *   elementName, // Filter by element[0].
+   *   elementType, // Instantiate as this subclass of RefractElement.
+   * }
+   *
+   * Returns an array of element objects.
+   */
+  contents(options={}) {
```

Comment:
There's probably a **lot** more work to be done here as it currently only supports simple arrays of elements we care about or simple hashes where a key contains such an array.